### ISSUE TITLE
Introduce built-in virtual machine manager Dragonball for Kata

### ISSUE THREAD

**Issue by studychao** [kata-containers/kata-containers#4257]
Dragonball Sandbox is a virtual machine manager(VMM) designed for container workloads that works with Linux Kernel Virtual Machine(KVM) to create lightweight virtual machines with industry-leading startup time, low CPU & memory resource consumption overhead, and concurrent startup optimization.

Dragonball Sandbox aims to provide a turnkey solution for the kata-containers community. We have done several optimizations for the container workload and we'll introduce them in detail with more documents and code in the coming weeks. Dragonball Sandbox has integrated with Kata 3.0 runtime as one process and gives users an out-of-the-box Kata experience without any complex virtualization version adaptation or configurations while providing strong performance and adequate virtualization service.

In conclusion, the two major advantages Dragonball offers to the Kata community are **built-in VMM experience** and **optimization for container workload scenarios**.

We plan to deliver Dragonball Sandbox as an optional VMM choice for the Kata community and the picture below is the design architecture of Dragonball.

![DragonballDesign](https://user-images.githubusercontent.com/31121654/168539835-f616758b-9855-4c37-bc60-e8e6896b15c1.png)
